### ISSUE TITLE
Error when using $count for $orderby, $filter and $expand in navigational properties

### ISSUE THREAD

**Issue by samuelgodbout** [OData/odata.net#809]
*Short summary (3-5 sentences) describing the issue.*  
$count does not seem supported in filter, expand or orderby

### Assemblies affected
OData.NET 7.X

### Reproduce steps
http://services.odata.org/V4/TripPinService/People?$orderby=Friends/$count
http://services.odata.org/V4/TripPinService/People?$filter=Friends/$count gt 0
http://services.odata.org/V4/TripPinService/People?$expand=Friends/$count

### Expected result
*What would happen if there wasn't a bug.*  
You should be able to see the count in the navigational properties

### Actual result
*What is actually happening.*  
Internal server error & bad request with the following error:
"message": "Found system token '$count' in select or expand clause 'Friends/$count'.",
"message": "The parent value for a property access of a property '$count' is not a single value. Property access can only be applied to a single value.",

**Comment by samuelgodbout** [comment]
https://github.com/OData/WebApi/issues/194
https://github.com/OData/odata.net/issues/440
https://github.com/OData/odata.net/issues/298

All these issues are closed however i still can't seem to get it working in my application nor in the example... am i doing something wrong?

**Comment by joshcomley** [comment]
If you call `ParseSelectAndExpand()` on `ODataQueryOptionParser` with a `$count` `$expand`, you get the error you've seen. It seems at the moment `odata.net` is unable to parse such expands.

Does anyone have an update on this? It's still the case with the week old 7.4 beta.

**Comment by rcreynolds53** [comment]
I am trying to do a count on an expand as well and still doesn't seem to be working. Any update on this?

**Comment by NetTecture** [comment]
Yeah. Any update? Running into the same here trying to put Odata into productio. Alternatives are adding it to a "Statistics" contained entity with all kinds of counts or actually using OData. Sadly this regression makes it unable to actually with counting navigational properties.

**Comment by Mortana89** [comment]
Same here, simple count of navigational properties are not working.

**Comment by robertmclaws** [comment]
@mikepizzo I kow the team has been working on expanded $count support across the board... I just tried to do an $expand=NavigationCollection/$count and it failed in 7.2.2. Any idea when this will be supported?

**Comment by Priyanka-Bharti-IBM** [comment]
I face this issue with SAP S4HANA where i get 501 not Implemented for below query

GET /sap/opu/odata4/sap/api_salesorder/srvd_a2x/sap/salesorder/0001/SalesOrder
    ?$expand=
        _Item(
            **$count=true;**
            $top=3;
            $select=
                DeliveryStatus,
                Product,
                RequestedQuantity,
                SalesOrderItem;
            $expand=
                _ItemRelatedObject(
                    **$count=true;**
                    $select=
                        SDDocRelatedObjectReference1,
                        SDDocRelatedObjectReference2,
                        SDDocRelatedObjectSequenceNmbr,
                        SDDocRelatedObjectSystem,
                        SDDocumentRelatedObjectType,
                        SalesOrder,
                        SalesOrderItem
                ),
                _ScheduleLine(
                    **$count=true;**
                    $select=
                        SalesOrder,
                        SalesOrderItem,
                        ScheduleLine,
                        ScheduleLineCategory,
                        ScheduleLineOrderQuantity
                )
        )
    &$select=
        SalesOrder,
        SoldToParty
    &$top=10
    &$skip=0
    &$count=true
